sourcegit/fed_mes_consume.py

```diff
@@ -0,0 +1,70 @@
+"""
+The code here handles receiving messages about events and has wrappers to process them.
+
+This module is meant to be imported in API and should be independent.
+"""
+import logging
+
+# not yet: https://github.com/fedora-infra/fedora-messaging/issues/111
+# from fedora_messaging import api
+
+import fedmsg
+
+
+logger = logging.getLogger(__name__)
+
+
+class Consumerino:
+    """
+    A class which provides an interface to consume messages via a callback
+    """
+    def __init__(self):
+        """
+
+        """
+        # timestamp = datetime.datetime.now().strftime("%Y%M%d-%H%M%S")
+        # self.binding = {
+        #     'exchange': 'amq.topic',  # The AMQP exchange to bind our queue to
+        #     'queue': f'source-git-{timestamp}',
+        #     'routing_keys': [topic],
+        # }
+
+    # def consume(self, callback):
+    #     logger.info("consuming messages on queue %s, routing keys = %s",
+    #                 self.binding["queue"], self.binding["routing_keys"])
+    #     api.consume(callback, self.binding)
+
+    @staticmethod
+    def iterate_gh_pulls() -> (str, str, dict):
+        """
+        Provide messages for all github pull-request-related events
+
+        Actions:
+            https://developer.github.com/v3/activity/events/types/#events-api-payload-28
+
+        :return: tuple, (full topic name, pull request action, dict with the message)
+        """
```

Comment:
We can move this to ogr in the future.

Reply:
Not really, we'll use ucho + frambo to receive messages and schedule jobs once our team open sources them.

Reply:
Yes, that sounds good.